README.md

```diff
@@ -1,2 +1,48 @@
-# vscode-ui-components
-Shared UI Components for CDT Cloud VS Code extensions
+# VSCode UI Components
+
+This is a VSCode components React library.
+
+## Project Structure
+
+```
+src/  
+│── base/           # Shared functionality  
+│── vscode/         # VSCode specific functionality  
+
+├── <component>/  
+│   ├── browser/    # React specific code  
+│   ├── common/     # Shared code between browser and VSCode  
+│   ├── vscode/     # VSCode integration (converters, webview providers, etc.)  
+
+├── <component>/  
+│   ├── *.tsx       # Components without VSCode integration  
+```
+
+## Installation
+
+`npm install @eclipse-cdt-cloud/vscode-ui-components`
+
+## Usage
```

Comment:
The original idea of this tree was to allow an easy path to reusing an existing `TreeDataProvider`. Can we add some notes here what a developer would need to do? For example where to use `getChildren` and `getTreeItem`, or how to map `TreeItem` to `CDTTreeItem`.